File: android/app/src/main/java/com/razinj/context_launcher/LauncherAppsCallback.java
package com.razinj.context_launcher;

import static com.razinj.context_launcher.Constants.PACKAGE_CHANGE_EVENT_IS_PACKAGE_REMOVED;
import static com.razinj.context_launcher.Constants.PACKAGE_CHANGE_EVENT_PACKAGE_NAME;
import static com.razinj.context_launcher.Constants.PACKAGE_CHANGE_INTENT_ACTION;

import android.content.Context;
import android.content.Intent;
import android.content.pm.LauncherApps;
import android.os.UserHandle;
import android.widget.Toast;

public class LauncherAppsCallback extends LauncherApps.Callback {
    private final Context context;

    public LauncherAppsCallback(Context context) {
        this.context = context;
    }

    @Override
    public void onPackageRemoved(String packageName, UserHandle user) {
        Toast.makeText(context, "onPackageRemoved: packageName: " + packageName, Toast.LENGTH_LONG).show();
        Intent intent = new Intent(PACKAGE_CHANGE_INTENT_ACTION);

        intent.putExtra(PACKAGE_CHANGE_EVENT_PACKAGE_NAME, packageName);
        intent.putExtra(PACKAGE_CHANGE_EVENT_IS_PACKAGE_REMOVED, true);

        context.sendBroadcast(intent);
    }

    @Override
    public void onPackageAdded(String packageName, UserHandle user) {
        Toast.makeText(context, "onPackageAdded: packageName: " + packageName, Toast.LENGTH_LONG).show();
        Intent intent = new Intent(PACKAGE_CHANGE_INTENT_ACTION);

        intent.putExtra(PACKAGE_CHANGE_EVENT_PACKAGE_NAME, packageName);
        intent.putExtra(PACKAGE_CHANGE_EVENT_IS_PACKAGE_REMOVED, false);

        context.sendBroadcast(intent);
    }

    @Override
    public void onPackageChanged(String packageName, UserHandle user) {
        Toast.makeText(context, "onPackageChanged: packageName: " + packageName, Toast.LENGTH_LONG).show();
        Intent intent = new Intent(PACKAGE_CHANGE_INTENT_ACTION);

        intent.putExtra(PACKAGE_CHANGE_EVENT_PACKAGE_NAME, packageName);
        intent.putExtra(PACKAGE_CHANGE_EVENT_IS_PACKAGE_REMOVED, false);

        context.sendBroadcast(intent);
    }

    @Override
    public void onPackagesAvailable(String[] packageNames, UserHandle user, boolean replacing) {
    }

    @Override
    public void onPackagesUnavailable(String[] packageNames, UserHandle user, boolean replacing) {
    }
}

File: android/app/src/main/java/com/razinj/context_launcher/LauncherAppsCallback.kt
package com.razinj.context_launcher

import android.content.pm.LauncherApps
import android.os.UserHandle

/**
 * Empty implementation of LauncherApps.Callback so we do not need to override all methods when
 * only parts of LauncherApps.Callback are needed.
 */
open class LauncherAppsCallback : LauncherApps.Callback() {
    override fun onPackageRemoved(packageName: String, user: UserHandle) {
    }

    override fun onPackageAdded(packageName: String, user: UserHandle) {
    }

    override fun onPackageChanged(packageName: String, user: UserHandle) {
    }

    override fun onPackagesAvailable(
        packageNames: Array<String>,
        user: UserHandle,
        replacing: Boolean
    ) {
    }

    override fun onPackagesUnavailable(
        packageNames: Array<String>,
        user: UserHandle,
        replacing: Boolean
    ) {
    }
}
